src/realm/cluster.cpp

```diff
@@ -56,10 +56,16 @@ void ClusterNode::IteratorState::init(State& s, ObjKey key)
     m_current_leaf.set_offset(m_key_offset);
 }
 
+const Table* ClusterNode::get_owning_table() const noexcept
+{
+    return m_tree_top.get_owning_table();
+}
+
 void ClusterNode::get(ObjKey k, ClusterNode::State& state) const
 {
     if (!k || !try_get(k, state)) {
-        throw KeyNotFound("No such object");
+        throw KeyNotFound(
+            util::format("No object with key '%1' in table '%2'", k.value, get_owning_table()->get_name()));
```

Comment:
This is much better than before, but it will expose internal table names with the "class_" prefix to the user. Could we remove that prefix if it's present? (not blocking)

Reply:
Also "table" is not a concept end-users is aware about. Perhaps just remove " table", then it will automatically be "class_ ..." :-)

Reply:
The same applies everywhere "table" is mentioned the other places.

Reply:
Removed the word "table" from error messages